ca/ca.go

```diff
@@ -514,17 +511,19 @@ func (ca *certificateAuthorityImpl) issuePrecertificateInner(ctx context.Context
 		return nil, nil, err
 	}
 
-	// Use the issuer which corresponds to the algorithm of the public key
-	// contained in the CSR, unless we have an allowlist of registration IDs
-	// for ECDSA, in which case switch all not-allowed accounts to RSA issuance.
+	// Select which pool of issuers to use, based on the to-be-issued cert's key
+	// type and whether we're using the ECDSA Allow List.
 	alg := csr.PublicKeyAlgorithm
 	if alg == x509.ECDSA && !features.Get().ECDSAForAll && ca.ecdsaAllowList != nil && !ca.ecdsaAllowList.permitted(issueReq.RegistrationID) {
 		alg = x509.RSA
 	}
-	issuer, ok := ca.issuers.byAlg[alg]
+
+	// Select a random issuer from among the active issuers of this key type.
+	issuerPool, ok := ca.issuers.byAlg[alg]
 	if !ok {
-		return nil, nil, berrors.InternalServerError("no issuer found for public key algorithm %s", csr.PublicKeyAlgorithm)
+		return nil, nil, berrors.InternalServerError("no issuers found for public key algorithm %s", csr.PublicKeyAlgorithm)
 	}
+	issuer := issuerPool[mrand.Intn(len(issuerPool))]
```

Comment:
The description on `mrand.Intn` says it will return a panic `if n <= 0`. On line 523 you could change it to read `if !ok || len(issuerPool) <= 0 {`. I get that if a map lookup doesn't succeed then `ok` will be `false` and trigger line 524. [This](https://github.com/letsencrypt/boulder/blob/327f96d281dd75b5b0ebeb0745b83a791fe361db/wfe2/wfe.go#L1737-L1739) [pattern](https://github.com/letsencrypt/boulder/blob/327f96d281dd75b5b0ebeb0745b83a791fe361db/wfe2/wfe.go#L932-L933) [is](https://github.com/letsencrypt/boulder/blob/327f96d281dd75b5b0ebeb0745b83a791fe361db/wfe2/verify.go#L294-L295) used in several other places, though the latter isn't such a good example.